packages/terra-form-select/src/Combobox.jsx

```diff
@@ -0,0 +1,231 @@
+  isInvalid: false,
+  noResultContent: undefined,
+  onChange: undefined,
+  onDeselect: undefined,
+  onSearch: undefined,
+  onSelect: undefined,
+  optionFilter: undefined,
+  placeholder: undefined,
+  required: false,
+  value: undefined,
+};
+
+class Combobox extends React.Component {
+  static defaultValue(props) {
+    if (props.value !== undefined) {
+      return null;
+    }
+
+    return (props.defaultValue !== undefined && props.defaultValue !== null) ? props.defaultValue : '';
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      tags: [],
+      value: Combobox.defaultValue(props),
+    };
+
+    this.display = this.display.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleDeselect = this.handleDeselect.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+  }
+
+  /**
+   * Returns the appropriate variant display
+   */
+  display() {
+    const selectValue = SelectUtil.value(this.props, this.state);
+
+    return SelectUtil.valueDisplay(this.props, selectValue);
+  }
+
+  /**
+   * Communicates changes to the value.
+   * @param {array|number|string} value - The value resulting from a change.
+   */
+  handleChange(value) {
+    if (this.props.value === undefined) {
+      this.setState({ value });
+    }
+
+    if (this.props.onChange) {
+      this.props.onChange(value);
+    }
+  }
+
+  /**
+   * Communicates the removal of a value from the selected options.
+   * @param {number|string} value - The value to be removed.
+   */
+  handleDeselect(value) {
+    this.handleChange(SelectUtil.deselect(this.props, this.state, value));
+
+    if (this.props.onDeselect) {
+      this.props.onDeselect(value);
+    }
+  }
+
+  /**
+   * Communicates the selection of a value.
+   * @param {number|string} value - The value of the selected option.
+   * @param {ReactNode} option - The selected option.
+   */
+  handleSelect(value, option) {
+    this.handleChange(SelectUtil.select(this.props, this.state, value));
+
+    // Add new tags for uncontrolled components.
+    if (this.props.value === undefined && !SelectUtil.findByValue(this.props, this.state, value)) {
+      this.setState(prevState => ({ tags: [...prevState.tags, <Option key={value} display={value} value={value} />] }));
+    }
+
+    if (this.props.onSelect) {
+      this.props.onSelect(value, option);
+    }
+  }
+
+  render() {
+    const {
+      allowClear, children, defaultValue, onChange, placeholder, required, value, intl, ...otherProps
+    } = this.props;
+
+    const defaultPlaceholder = intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+    const selectPlaceholder = placeholder === undefined ? defaultPlaceholder : placeholder;
+    let clearOptionDisplay;
+
+    if (allowClear) {
+      if (selectPlaceholder.length === 0) {
+        clearOptionDisplay = defaultPlaceholder;
+      } else {
+        clearOptionDisplay = selectPlaceholder;
+      }
+    }
+
+    return (
+      <Frame
```

Comment:
Do we have an issue logged to flatten out the Frame?

Reply:
Not logged yet. Was going to log one immediately after this merges, but I can go ahead and just ref this PR

Reply:
Logged here: https://github.com/cerner/terra-core/issues/2582